Tests/Singulink.Globalization.Currency.Tests/ImmutableSortedMoneySetTests/RemoveTests.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System.Collections.Generic;
+using System.Collections.Immutable;
+using Shouldly;
+
+namespace Singulink.Globalization.Tests.ImmutableSortedMoneySetTests;
+
+[TestClass]
+public class RemoveTests
+{
+    private static readonly Money Usd100 = new(100m, "USD");
+    private static readonly Money Cad50 = new(50m, "CAD");
+    private static readonly Money Eur25 = new(25m, "EUR");
+    private static readonly ImmutableSortedMoneySet _immutableSet = [Usd100, Cad50, Eur25];
+
+    // public bool Remove(string currencyCode) tests
+
+    [TestMethod]
+    public void RemoveCurrencyCode_CurrencyFound_RemovesValueAndReturnsTrue()
+    {
+        var updatedSet = _immutableSet.Remove("USD");
+        updatedSet.Count.ShouldBe(2);
+        updatedSet.ShouldBe([Cad50, Eur25]);
+    }
+
+    [TestMethod]
+    public void RemoveCurrencyCode_CurrencyNotFound_ReturnsFalseAndNoChange()
+    {
+        var updatedSet = _immutableSet.Remove("JPY");
```

Comment:
The set may or may not be updated in these operations, i.e. in this case the returned set should be the same reference as the set the operation is called on, so perhaps "var resultSet` is a better name for the variable that holds what is returned from immutable set methods in all the test methods.

For operations that are expected to have no change, assert that the references of the input set and result set are the same, i.e. `resultSet.ShouldBeSameAs(Set);`,  as the last assertion.